src/Workspaces/Core/Portable/Storage/SQLite/Interop/SqlConnection.cs

```diff
@@ -129,7 +129,7 @@ public ResettableSqlStatement GetResettableStatement(string query)
             return new ResettableSqlStatement(statement);
         }
 
-        public void RunInTransaction(Action action)
+        public TResult RunInTransaction<T, TResult>(Func<T, TResult> action, T userData)
```

Comment:
note: type-inference miht work if you pass the data first, then the func.  That would be easier to read too as the func is often multi-line, so it's difficult to understand what's being passed in.

Reply:
Sure, I can change that.